### PR TITLE
Add test for payment complete

### PR DESCRIPTION
**Pull request by v18**
This PR updates the tests written for `process_payment` to include a test for the `payment_complete` call for a successful intent. To do that, the WC_Order object is mocked instead of being created using the helper class. This makes it easy to check whether `$order->payment_complete()` was called.

To facilitate this, the logic of the function `process_payment` is moved to another function called `payment_complete_for_order`, which takes an order object instead of an order ID. This is needed because it wasn't possible to mock the global function `wc_get_order` which gets the order using the order ID. I tried mocking it using WP_Mock and ran into some trouble getting it set up. I think it's still worthwhile to try again later, but I wanted to get these tests out the door for now.

Big thanks to @jrodger for helping me think through the architecture and how we can ship this sooner rather than later.

I also added an assertion for each test to check whether or not the cart was emptied. This is done by passing in `WC()->cart` to `payment_complete_for_order`.

There are only two things left untested after the intent object is fetched:
1. reducing stock
2. a zero-amount order

Testing reducing the stock will need to come with setting up something like WP_Mock since it's another global function. I intend to add tests for the zero-amount order soon, as well as tests for `update_order_status()` which will be similar to what we already have for `process_payment`.

Related PRs: #606, #686 

### Testing instructions

Run `./vendor/bin/phpunit`.

### CODE REVIEW

**Comment by v18** [comment]
Hey @jrodger, I updated this PR with these changes:

- mocking the order object in all tests
- passing in `WC()->cart` so that we can test if the cart was emptied

The commits are organized by test, I tried to make it easier to see at a glance each test's changes. I'll add more information in the main description.

File: tests/test-class-wc-payment-gateway-wcpay-process-payment.php [hunk 1]
```diff
@@ -97,16 +103,29 @@ public function setUp() {
 	}
 
 	/**
+	 * Test processing payment with the status 'succeeded.
```
[the remark this answers is not in the document]
Comment (v18) [reply tests/test-class-wc-payment-gateway-wcpay-process-payment.php]: Thanks! Fixed.
